### PR TITLE
Run e2e locally

### PR DESCRIPTION
**Pull request by chatton**
This PR adds the ability to run e2e tests locally (using go test)

The script added sets up all the required environment variables that the testing environment requires. These are read from the config file.

### All Submissions:

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### CODE REVIEW

**Comment by irajdeep** [comment]
Will running the `e2e` test locally be the only option now?  I mean is there a make target to run the e2e against a K8s cluster?

File: scripts/dev/dev_config.py [hunk 1]
```diff
@@ -74,6 +74,25 @@ def version_upgrade_hook_image(self) -> str:
     def readiness_probe_image(self) -> str:
         return self._config["readiness_probe_image"]
 
+    # these directories are used from within the E2E tests when running locally.
+    @property
+    def role_dir(self) -> str:
+        if "role_dir" in self._config:
+            return self._config["role_dir"]
+        return os.getcwd() + "/config/rbac"
```
Comment (rodrigovalin) [inline scripts/dev/dev_config.py:82]: use `os.path.join` instead of string concatenation

File: test/e2e/setup/setup.go [hunk 2]
```diff
@@ -30,13 +32,13 @@ import (
...
-	deployDir      = "/workspace/config/manager"
-	roleDir        = "/workspace/config/rbac"
+	performCleanupEnv = "PERFORM_CLEANUP"
+	deployDirEnv      = "DEPLOY_DIR"
+	roleDirEnv        = "ROLE_DIR"
 )
 
 func Setup(t *testing.T) *e2eutil.Context {
-	ctx, err := e2eutil.NewContext(t, os.Getenv(performCleanup) == "True")
+	ctx, err := e2eutil.NewContext(t, os.Getenv(performCleanupEnv) == "True")
```
Comment (rodrigovalin) [inline test/e2e/setup/setup.go:41]:
```suggestion
	ctx, err := e2eutil.NewContext(t, strings.ToLower(os.Getenv(performCleanupEnv)) == "true")
```

**Review by rodrigovalin — APPROVED** [review]
2 small comments

LGTM

**Comment by chatton** [comment]
@irajdeep yes the `e2e-k8s` will deploy the e2e application and run everything in kubernetes.

File: test/e2e/setup/setup.go [hunk 3]
```diff
@@ -30,13 +31,13 @@ import (
...
-	deployDir      = "/workspace/config/manager"
-	roleDir        = "/workspace/config/rbac"
+	performCleanupEnv = "PERFORM_CLEANUP"
+	deployDirEnv      = "DEPLOY_DIR"
+	roleDirEnv        = "ROLE_DIR"
 )
 
 func Setup(t *testing.T) *e2eutil.Context {
-	ctx, err := e2eutil.NewContext(t, os.Getenv(performCleanup) == "True")
+	ctx, err := e2eutil.NewContext(t, envvar.ReadBool(performCleanupEnv))
```
Comment (chatton) [inline test/e2e/setup/setup.go:40]: @rodrigovalin  we actually have a function that handles this already (including lower casing)

**Pull request merged**